arcade/examples/sound_demo.py

```diff
@@ -93,140 +88,43 @@ def setup(self):
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=1, volume=volume)
-        button.center_x = SCREEN_WIDTH - BUTTON_SIZE
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        y = SCREEN_HEIGHT / 2
-        volume = 0.5
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=-1.0, volume=volume)
-        button.center_x = BUTTON_SIZE
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=-0.5, volume=volume)
-        button.center_x = SCREEN_WIDTH / 4
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=0, volume=volume)
-        button.center_x = SCREEN_WIDTH / 2
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=0.5, volume=volume)
-        button.center_x = SCREEN_WIDTH / 4 * 3
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=1, volume=volume)
-        button.center_x = SCREEN_WIDTH - BUTTON_SIZE
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        y = SCREEN_HEIGHT / 2 - 50
-        volume = 1
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=-1.0, volume=volume)
-        button.center_x = BUTTON_SIZE
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=-0.5, volume=volume)
-        button.center_x = SCREEN_WIDTH / 4
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=0, volume=volume)
-        button.center_x = SCREEN_WIDTH / 2
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=0.5, volume=volume)
-        button.center_x = SCREEN_WIDTH / 4 * 3
-        button.center_y = y
-        self.button_sprites.append(button)
-
-        button = SoundButton(":resources:sounds/upgrade4.wav", pan=1, volume=volume)
-        button.center_x = SCREEN_WIDTH - BUTTON_SIZE
-        button.center_y = y
-        self.button_sprites.append(button)
+        upgrade_sound = ":resources:sounds/upgrade4.wav"
+        volume_variation = [0.1, 0.5, 1]
+        y_offset = [50, 0, -50]
+        button_x_spread = [
+            BUTTON_SIZE,
+            SCREEN_WIDTH / 4,
+            SCREEN_WIDTH / 2,
+            SCREEN_WIDTH / 4 * 3,
+            SCREEN_WIDTH - BUTTON_SIZE,
+        ]
+
+        for vol, y_offset in zip(volume_variation, y_offset):
+            for pan_setting, x_pos in zip(SOUND_PANNING, button_x_spread):
```

Comment:
Although this `zip`-based approach might confuse some beginners, it's very concise and I think I like the idea. :+1:

It'd be helpful for beginners if you add some comments with the following:
1. A few words explaining what `zip` does
2. A link to the python doc for `zip`: https://docs.python.org/3/library/functions.html#zip

Maybe something like this?
```py
# Arrange the buttons in a grid
# The zip function puts pieces of two iterables together
# For more information, see the python doc:
# https://docs.python.org/3/library/functions.html#zip
```